File: Programmers/Lv3/등굣길.py
def solution(M, N, puddles):
    roads = [[0 for _ in range(M+1)] for _ in range(N+1)]   #dp

    for puddle in puddles:
        roads[puddle[1]][puddle[0]] = -1           #puddle 위치 표시

    for i in range(1,N+1):
        for j in range(1,M+1):
            if i==1 and j==1:
                roads[1][1] = 1                    #시작 = 1
            elif roads[i][j] == -1:                 # puddle일경우 0으로 바꾸기
                roads[i][j] = 0
            else:                                  #dp 더하기
                roads[i][j] = roads[i-1][j]+roads[i][j-1]
    return roads[N][M] % 1000000007

File: Programmers/Lv3/등굣길.js
function solution(m, n, puddles) {
  // 1. n x m 크기의 배열을 1로 초기화
  let dp = Array.from(Array(n + 1), () => Array(m + 1).fill(0));
  dp[1][1] = 1;
  // 2. 웅덩이 처리
  for (let [x, y] of puddles) {
    dp[y][x] = -1; // 웅덩이 위치는 0으로 설정
  }

  // 3. 경로 계산
  for (let i = 1; i <= n; i++) {
    for (let j = 1; j <= m; j++) {
      if (i === 1 && j === 1) continue;
      if (dp[i][j] === -1) {
        dp[i][j] = 0;
      } else {
        dp[i][j] = (dp[i - 1][j] + dp[i][j - 1]) % 1000000007;
      }
    }
  }

  return dp[n][m]; // 마지막 칸의 값 반환
}
